### PR TITLE
feat(cloud): `GET /api/tenants`

### PR DESCRIPTION
**Pull request by gao-sun**
## Summary

- init router and queries
- add `ignorePathnames` option to SPA middleware functions to filter out API requests
- add `GET /api/tenants` API to fetch all available tenant info for the current user
- admin console become dedicated SPA for cloud, mount it under the root path `/`
- dynamically fetch tenant ID from URL in console

### To-dos

- add `POST /api/tenants` for onboarding
- add `withAuth()` to auth requests via Access Token
- add cloud `EnvSet` to aggregate env variables

## Testing

- `http://localhost:3003` loads admin console correctly
- `GET http://localhost:3003/api/tenants` returns an array of available tenant info

### CODE REVIEW

File: packages/cloud/src/routes/tenants.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { createTenantsQueries } from '#src/queries/tenants.js';
+import { getTenantIdFromManagementApiIndicator } from '#src/utils/tenant.js';
+
+const { getManagementApiLikeIndicatorsForUser } = createTenantsQueries(client);
+
+export const tenants = new Router('/tenants').get(
+  '/',
+  { response: z.object({ id: z.string(), indicator: z.string() }).array() },
+  async (context, next) => {
+    const { rows } = await getManagementApiLikeIndicatorsForUser('some_user_id');
```
Comment (charIeszhao) [inline packages/cloud/src/routes/tenants.ts:14]: Is this on purpose? `some_user_id` looks strange here

Comment (gao-sun) [reply packages/cloud/src/routes/tenants.ts:14]: it's for test and on purpose, will be replaced with the result of auth soon

File: packages/cloud/src/utils/postgres.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  } = Object.fromEntries(url.searchParams);
+
+  if (Object.keys(unsupportedOptions).length > 0) {
+    console.warn(
+      {
+        unsupportedOptions,
+      },
+      'unsupported DSN parameters'
+    );
+  }
```
Comment (charIeszhao) [inline packages/cloud/src/utils/postgres.ts:51]: Wouldn't it be better if we do this check earlier, like in the beginning of this function?

**Review by charIeszhao — APPROVED** [review]

File: packages/cloud/src/utils/postgres.ts (hunk 2 above)
Comment (gao-sun) [reply packages/cloud/src/utils/postgres.ts:51]: sure i didn't notice this. just copied and edited for our usage

**Review by wangsijie — APPROVED** [review]

File: packages/cloud/src/utils/postgres.ts (hunk 2 above)
Comment (gao-sun) [reply packages/cloud/src/utils/postgres.ts:51]: took a look at the function again, looks like the order is reasonable by "importance of the option"

Comment (gao-sun) [reply packages/cloud/src/utils/postgres.ts:51]: discussed offline

**Pull request merged**